### PR TITLE
feat(dashpay): use mock state repository

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

This will open up DashPay functions by "faking it until we make it".  Mocking is done in the `dashpay` libraries

## Related PR's and Dependencies

https://github.com/dashpay/android-dpp/pull/40
https://github.com/dashpay/android-dashpay/pull/73
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/build.gradle [hunk 1]
```diff
@@ -50,8 +50,8 @@ dependencies {
     implementation "androidx.biometric:biometric:1.1.0"
 
     implementation "org.dashj:dashj-core:$dashjVersion"
-    implementation 'org.dashj.platform:dashpay:0.24-SNAPSHOT'
-    implementation 'org.dashj.platform:platform-core:0.24-SNAPSHOT'
+    implementation 'org.dashj.platform:dashpay:0.24-MOCK-SNAPSHOT'
+    implementation 'org.dashj.platform:platform-core:0.24-MOCK-SNAPSHOT'
```
Comment (HashEngineering) [inline wallet/build.gradle:54]: All data mocking is handled here.  Any data created in DashPay won't be saved there.  I didn't want to add more code to this PR requiring changes to DB tables, etc.

For now these changes will allow a username to be created or restored.

File: wallet/src/de/schildbach/wallet/service/CoinJoinService.kt [hunk 2]
```diff
@@ -210,9 +211,12 @@ class CoinJoinMixingService @Inject constructor(
             wallet: WalletEx?,
             sessionId: Int,
             denomination: Int,
+            state: PoolState?,
             message: PoolMessage?,
+            address: MasternodeAddress?,
+            joined: Boolean
         ) {
-            super.onSessionComplete(wallet, sessionId, denomination, message)
+            super.onSessionComplete(wallet, sessionId, denomination, state, message, address, joined)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/service/CoinJoinService.kt:219]: These are new 19.1-CJ-SNAPSHOT changes, but I haven't published it yet.

File: wallet/src/de/schildbach/wallet/service/platform/PlatformSyncService.kt [hunk 3]
```diff
@@ -194,7 +194,7 @@ class PlatformSynchronizationService @Inject constructor(
     override suspend fun updateContactRequests() {
 
         // if there is no wallet or identity, then skip the remaining steps of the update
-        if (platformRepo.hasIdentity || walletApplication.wallet == null) {
+        if (!platformRepo.hasIdentity || walletApplication.wallet == null) {
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/service/platform/PlatformSyncService.kt:197]: These are some bugs leftover from merging Core 19 changes, masternode changes, etc -- but we didn't actually test this to see if it was working.

File: wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt [hunk 4]
```diff
@@ -216,6 +217,8 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
             dashPayProfileDao.loadAll().forEach {
                 platform.stateRepository.addValidIdentity(it.userIdentifier)
             }
+
+            platform.stateRepository.storeIdentity(blockchainIdentity.identity!!)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt:221]: This is where we could add other documents that we created, but at this point it doesn't seem necessary to modify DB tables to save those documents.

File: wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt [hunk 5]
```diff
@@ -1019,12 +1022,7 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
         val firstIdentityKey = getBlockchainIdentityKey(0, encryptionKey) ?: return null
 
         return try {
-            val identityBytes = platform.client.getIdentityByFirstPublicKey(firstIdentityKey.pubKeyHash, true)
-            if (identityBytes != null && identityBytes.isNotEmpty()) {
-                platform.dpp.identity.createFromBuffer(identityBytes)
-            } else {
-                null
-            }
+            platform.stateRepository.fetchIdentityFromPubKeyHash(firstIdentityKey.pubKeyHash)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt:1025]: as in Android DashPay Library, `platform.stateRepository.` is the better way to make platform calls, so the "intercepter" will catch the calls.

File: wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt [hunk 6]
```diff
@@ -656,6 +656,7 @@ class CreateIdentityService : LifecycleService() {
         blockchainIdentityData = BlockchainIdentityData(CreationState.NONE, null, null, null, true)
 
         val authExtension = walletApplication.wallet!!.getKeyChainExtension(AuthenticationGroupExtension.EXTENSION_ID) as AuthenticationGroupExtension
+        //authExtension.setWallet(walletApplication.wallet!!) // why is the wallet not set?  we didn't deserialize it probably!
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt:659]: This line is in danger of being abandoned here forever. Maybe `TODO` should be added?

**Review by Syn-McJ — APPROVED** [review]
Looks good

**Review by samettek — APPROVED** [review]

File: wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt (hunk 6 above)
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt]: Thanks.  This line will be removed as the bug was fixed elsewhere.

**Pull request merged**